crates/db_schema/src/impls/local_user.rs

```diff
@@ -216,6 +217,75 @@ impl LocalUser {
       blocked_instances,
     })
   }
+
+  /// Checks to make sure the acting moderator is higher than the target moderator
+  pub async fn is_higher_admin_check(
+    pool: &mut DbPool<'_>,
+    admin_person_id: PersonId,
+    target_person_ids: &[PersonId],
+  ) -> Result<(), Error> {
+    let conn = &mut get_conn(pool).await?;
+
+    // Build the list of persons
+    let mut persons = target_person_ids.to_owned();
+    persons.push(admin_person_id);
+    persons.dedup();
+
+    let res = local_user::table
+      .filter(local_user::admin.eq(true))
+      .filter(local_user::person_id.eq_any(persons))
+      .order_by(local_user::id)
+      // This does a limit 1 select first
+      .first::<LocalUser>(conn)
+      .await?;
+
+    // If the first result sorted by published is the acting mod
+    if res.person_id == admin_person_id {
+      Ok(())
+    } else {
+      Err(diesel::result::Error::NotFound)
```

Comment:
Should return `NotHigherAdmin`.

Reply:
Why only diesel errors? LemmyError is available here so we might as well use it directly.

Reply:
Its the standard for every DB call, and every function in db_schema. Other functions then use the `with_lemmy_type` to convert that error.

If we want to change that to make every DB call return lemmy errors, that should be done in a separate ticket.

Reply:
Im not suggesting to change it everywhere, only in this specific case where we actually need a LemmyResult.

Reply:
To solve this problem without changing the error type, return `Ok(false)` here

Reply:
Ok this is done now.